checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallInferenceLogic.java

```diff
@@ -43,104 +82,596 @@ public class MustCallInferenceLogic {
   /** The control flow graph. */
   private final ControlFlowGraph cfg;
 
+  /** The MustCallConsistencyAnalyzer. */
+  private final MustCallConsistencyAnalyzer mcca;
+
+  /** The MethodTree of the cfg. */
+  private MethodTree enMethodTree;
+
+  /** The element for the enMethodTree. */
+  private ExecutableElement enMethodElt;
+
   /**
    * Creates a MustCallInferenceLogic. If the type factory has whole program inference enabled, its
    * postAnalyze method should instantiate a new MustCallInferenceLogic using this constructor and
    * then call {@link #runInference()}.
```

Comment:
Why does the client need to make two calls? It would be cleaner if either:
* this constructor calls `runInference()` (but that might not be desirable, since it will make this class hard to reuse), or
* this class provided a static factory method that takes the same parameters as this constructor, calls the constructor, and then calls `runInference()`. Then, clients could just use that method.

Reply:
Of these two, I definitely prefer the second.

Reply:
Thanks for the suggestion. I have implemented the second option..